File: Valid Sudoku/Valid_Sudoku.java
class Solution {
    private int row = 0;
    private int column = 0;
    private HashSet<Character> subBox1 = new HashSet<>();
    private HashSet<Character> subBox2 = new HashSet<>();
    private HashSet<Character> subBox3 = new HashSet<>();
    private HashSet<Character> subBox4 = new HashSet<>();
    private HashSet<Character> subBox5 = new HashSet<>();
    private HashSet<Character> subBox6 = new HashSet<>();
    private HashSet<Character> subBox7 = new HashSet<>();
    private HashSet<Character> subBox8 = new HashSet<>();
    private HashSet<Character> subBox9 = new HashSet<>();
    
    public boolean isValidSudoku(char[][] board) {
        while (row <= 8 && column <= 8){
            char dig = board[row][column];
            HashSet<Character> set = getSetBasedOnXY(row, column);
            if (dig != '.'){
                if (set.contains(dig)){
                    return false;
                }
                set.add(dig);
                int left = column-1;
                int right = column+1;
                while (left >= 0){
                    if (dig == board[row][left]){
                        return false;
                    }
                    left--;
                }
                while (right <= 8){
                    if (dig == board[row][right]){
                        return false;
                    }
                    right++;
                }
                int up = row-1;
                int down = row+1;
                while (up >= 0){
                    if (dig == board[up][column]){
                        return false;
                    }
                    up--;
                }
                while (down <= 8){
                    if (dig == board[down][column]){
                        return false;
                    }
                    down++;
                }
            }
            if (column == 8){
                row++;
                column = 0;
            }
            else {
                column++;
            }
        }
        return true;
    }
    
    private HashSet<Character> getSetBasedOnXY(int row, int column){
        if ( row <= 2 && column <= 2 ){
            return subBox1;
        } else if ( row <= 2 && (column > 2 && column <= 5) ){
            return subBox2;
        } else if ( row <= 2 && (column > 5 && column <= 8) ){
            return subBox3;
        } else if ( (row > 2 && row <= 5) && column <= 2 ){
            return subBox4;
        } else if ( (row > 2 && row <= 5) && (column > 2 && column <= 5) ){
            return subBox5;
        } else if ( (row > 2 && row <= 5) && (column > 5 && column <= 8) ){
            return subBox6;
        } else if ( (row > 5 && row <= 8) && column <= 2 ){
            return subBox7;
        } else if ( (row > 5 && row <= 8) && (column > 2 && column <= 5) ){
            return subBox8;
        } else {
            return subBox9;
        }
    }
}

File: Coding Problems & My Solutions/Valid Sudoku/Valid_Sudoku.swift
class Solution {
    var subBox1: Set<Character> = []
    var subBox2: Set<Character> = []
    var subBox3: Set<Character> = []
    var subBox4: Set<Character> = []
    var subBox5: Set<Character> = []
    var subBox6: Set<Character> = []
    var subBox7: Set<Character> = []
    var subBox8: Set<Character> = []
    var subBox9: Set<Character> = []
    func getSubBoxNo(_ i: Int, _ j: Int) -> Int {
        if i < 3 && j < 3 {
            return 1
        }
        else if i < 3 && (j >= 3 && j < 6) {
            return 2
        }
        else if i < 3 && j >= 6 {
            return 3
        }
        else if (i >= 3 && i < 6) && j < 3 {
            return 4
        }
        else if (i >= 3 && i < 6) && (j >= 3 && j < 6) {
            return 5
        }
        else if (i >= 3 && i < 6) && j >= 6 {
            return 6
        }
        else if i >= 6 && j < 3 {
            return 7
        }
        else if i >= 6 && (j >= 3 && j < 6) {
            return 8
        } else {
            return 9
        }
    }
    func isValidSudoku(_ board: [[Character]]) -> Bool {
        for (i, row) in board.enumerated() {
            for (j, cell) in row.enumerated() {
                if board[i][j] == "." {
                    continue
                }
                // Check if Valid Row
                var a = 0
                while a < row.count {
                    if a != j && board[i][a] == board[i][j] {
                        return false
                    }
                    a += 1
                }
                // Check if Valid Col
                var b = 0
                while b < board.count {
                    if b != i && board[b][j] == board[i][j] {
                        return false
                    }
                    b += 1
                }
                // Check if Valid SubBox
                switch getSubBoxNo(i, j) {
                    case 1:
                        if subBox1.contains(board[i][j]) {
                            return false
                        }
                        subBox1.insert(board[i][j])
                    case 2:
                        if subBox2.contains(board[i][j]) {
                            return false
                        }
                        subBox2.insert(board[i][j])
                    case 3: 
                        if subBox3.contains(board[i][j]) {
                            return false
                        }
                        subBox3.insert(board[i][j])
                    case 4:
                        if subBox4.contains(board[i][j]) {
                            return false
                        }
                        subBox4.insert(board[i][j])
                    case 5:
                        if subBox5.contains(board[i][j]) {
                            return false
                        }
                        subBox5.insert(board[i][j])
                    case 6:
                        if subBox6.contains(board[i][j]) {
                            return false
                        }
                        subBox6.insert(board[i][j])
                    case 7:
                        if subBox7.contains(board[i][j]) {
                            return false
                        }
                        subBox7.insert(board[i][j])
                    case 8:
                        if subBox8.contains(board[i][j]) {
                            return false
                        }
                        subBox8.insert(board[i][j])
                    default:
                        if subBox9.contains(board[i][j]) {
                            return false
                        }
                        subBox9.insert(board[i][j])
                }
            }
        }
        return true
    }
}